[doc][tune] Add tune checkpoint user guide. (#33145)
ray-project/ray: doc/source/tune/api/schedulers.rst

> * If you are using the functional API for tuning, the current trial resources can be

Reviewer: ```suggestion
* If you are using the functional API for tuning, get the current trial resources
```
